src/BinFile/MachHelper.fs

```diff
@@ -76,6 +76,13 @@ let isNXEnabled mach =
   not (mach.MachHdr.Flags.HasFlag MachFlag.MHAllowStackExecution)
   || mach.MachHdr.Flags.HasFlag MachFlag.MHNoHeapExecution
 
+let getBaseAddr (mach: Mach) =
+  let textSegment =
+    mach.Segments
+    |> List.filter (fun seg -> seg.FileOff = 0UL && seg.FileSize > 0UL)
```

Comment:
Is this check `seg.FileOff = 0UL`  safe? I am not sure if this will work for fat binaries for example. Why don't we just do the same as in ELF: sort by VMAddr and pick the minimum?

Reply:
**Note**: I'm not an expert in `Mach-O` binaries analysis, so Please let me know if I'm wrong :).
1- FAT binaries aka Universal binaries are just an archive files that contain multiple Mach-O files, such files are handled by `MachParser` [here](https://github.com/B2R2-org/B2R2/blob/d3124ffad73986b5973c724df700bad9918f628a/src/BinFile/MachParser.fs#L86) by extracting an object file based on the chosen `ISA`, so we are not interested in such Format, here is a quote from Apple's documentation:

> Binaries that contain object files for more than one architecture are not Mach-O files. They archive one or more Mach-O files.

2- In my code I focused on Segments that have an actual existance on-disk (i.e: `filesize > 0`), runtime-allocated Segments like `__PAGEZERO` are discarded, so I thought we can safely assume that "on-disk Segments" are impossible to overlap (i.e: have two or more segments point to the same file offset), therefore we can make sure that the segment at offset 0 is the base segment. here is what Apple's documentation states:

> The header and load commands are considered part of the first segment of the file for paging purposes. In an
> executable file, this generally means that the headers and load commands live at the start of the `__TEXT`
> segment because that is the first segment that contains data. The `__PAGEZERO` segment contains no data on
> disk, so it’s ignored for this purpose.

3- On the other hand, I think you're right if you check both `VMAddr` and `fileSize`, otherwise you'll end-up picking the wrong segement address, so it's up to you to decide in the end.

Reply:
I think it would be safer if we use `minBy` as in ELF, instead of just picking the head. Can you modify it?

Reply:
Done.